src/components/DocumentsPage/components/DocumentsEditor/index.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import './styles.css';
+
+class DocumentsEditor extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { id: null, title: "", content: "" };
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.document && nextProps.document.title !== this.state.title) {
+      this.setState({ ...nextProps.document  });
+    }
+  }
+
+  defaultState() {
+    this.setState({ id: null, title: "", content: "" })
+  };
+
+  updateDocument = (event) => {
+    var target = event.target;
+    this.props.onChange(target.name, target.value)
```

Comment:
This can be buggy - it'd be better not to keep the state at all, but handle everything via props in this case - then you wouldn't need to setState inside of componentWillReceiveProps